tensorboard/util.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TensorBoard helper routine module.
+
+This module is a trove of succinct generic helper routines that don't
+pull in any heavyweight dependencies aside from TensorFlow.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import locale
+
+
+def closeable(klass):
+  """Makes a class with a close method able to be a context manager.
+
+  This decorator is a great way to avoid having to choose between the
+  boilerplate of __enter__ and __exit__ methods, versus the boilerplate
+  of using contextlib.closing on every with statement.
+
+  Args:
+    klass: The class being decorated.
+
+  Raises:
+    ValueError: If class didn't have a close method, or already
+        implements __enter__ or __exit__.
+  """
+  if 'close' not in klass.__dict__:
+    # coffee is for closers
+    raise ValueError('Class does not define a close() method: ' + klass)
+  if '__enter__' in klass.__dict__ or '__exit__' in klass.__dict__:
+    raise ValueError('Class already defines __enter__ or __exit__: ' + klass)
+  klass.__enter__ = lambda self: self
+  klass.__exit__ = lambda self, exc_type, exc_val, exc_tb: self.close()
```

Comment:
Is it intended behavior that if `self.close()` returns truthy then the exception is swallowed? (I'd argue that it shouldn't be, and if it should be than at very least it should be documented.) See "Specification: the `with` statement" in [PEP343](https://www.python.org/dev/peps/pep-0343/), or try:
```py
>>> @closeable
... class Kirby(object):
...     def __init__(self, should_swallow):
...         self._should_swallow = should_swallow
...     def close(self):
...         print "Hiiii"
...         return self._should_swallow
... 
>>> with Kirby(should_swallow=False): raise ValueError("hyah ei")
... 
Hiiii
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: hyah ei
>>> with Kirby(should_swallow=True): raise ValueError("hyah ei")
... 
Hiiii
```